File: docker/Dockerfile.php
# PHP Leaderboard Service
FROM php:8.2-cli

RUN apt-get update && apt-get install -y \
    git \
    unzip \
    && rm -rf /var/lib/apt/lists/*

# Install Composer
COPY --from=composer:latest /usr/bin/composer /usr/bin/composer

WORKDIR /app
COPY backend/php/composer.json .
RUN composer install --no-dev --optimize-autoloader

COPY backend/php/ .

EXPOSE 8083

CMD ["php", "-S", "0.0.0.0:8083", "index.php"]

File: docker/Dockerfile.cpp
# C++ Sudoku Solver Service
FROM ubuntu:22.04

RUN apt-get update && apt-get install -y \
    build-essential \
    cmake \
    pkg-config \
    libjsoncpp-dev \
    libcurl4-openssl-dev \
    && rm -rf /var/lib/apt/lists/*

WORKDIR /app
COPY backend/cpp/ .

RUN mkdir build && cd build && \
    cmake .. && \
    make

EXPOSE 8081

CMD ["./build/bin/sudoku_solver"]